### PR TITLE
Updates for building with containers

### PR DESCRIPTION
**Pull request by mark-a-potts**
Updates for container modules and detecting a container build.

### CODE REVIEW

File: modulefiles/build.container.intel.lua
Comment (DavidBurrows-NCO) [inline modulefiles/build.container.intel.lua:1]: @mark-a-potts Can you change Hera to container (or whatever you're plan is)?  Do you need the path to spack-stack-1.8.0 in here or can it be removed?

Comment (DavidBurrows-NCO) [reply modulefiles/build.container.intel.lua:1]: How are you handling fix files at `fix/link_fixdirs.sh`.
The command looks like `./link_fixdirs.sh emc $machine`

Comment (mark-a-potts) [reply modulefiles/build.container.intel.lua:1]: I did not address that. I don't think we want to include the fix files in the container, so I am not sure if we can just leave it as is, or need to add a warning or something to create a FIX directory locally and point to it with an environment variable.

Comment (DavidBurrows-NCO) [reply modulefiles/build.container.intel.lua:1]: Sounds good. We'll see what George says.

**Pull request merged**